### PR TITLE
Expose buildTreePath on Project and ProjectComponentIdentifier

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test rFM

**Review by reinsch82 — APPROVED** [review]
LGTM

**Review by wolfs — CHANGES_REQUESTED** [review]
The API changes look good, though tests are missing, especially for ProjectComponentIdentifier.buildTreePath.

File: subprojects/core/src/test/groovy/org/gradle/api/internal/project/DefaultProjectStateRegistryTest.groovy [hunk 1]
```diff
@@ -53,6 +53,7 @@ class DefaultProjectStateRegistryTest extends ConcurrentSpec {
         root.identityPath == Path.ROOT
         root.projectPath == Path.ROOT
         root.componentIdentifier.projectPath == ":"
+        root.componentIdentifier.buildTreePath == ":"
```
[the remark this answers is not in the document]
Comment (donat) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/project/DefaultProjectStateRegistryTest.groovy:56]: I've adjusted the tests you mentioned.

**Review by wolfs — APPROVED** [review]

**Comment by donat** [comment]
@bot-gradle test and merge please

**Pull request merged**